src/status_im/transport/message/transit.cljs

```diff
@@ -44,14 +44,36 @@
   (rep [this {:keys [name profile-image address fcm-token]}]
     #js [name profile-image address fcm-token]))
 
+;; It's necessary to support old clients understanding only older, needlesy complex and verbose command content
+(defn- new->legacy-command-data [{:keys [command-path params] :as content}]
+  (get {["send" #{:personal-chats}]    [{:command-ref ["transactor" :command 83 "send"]
+                                         :command "send"
+                                         :bot "transactor"
+                                         :command-scope-bitmask 83}
+                                        constants/content-type-command]
+        ["request" #{:personal-chats}] [{:command-ref ["transactor" :command 83 "request"]
+                                         :request-command-ref ["transactor" :command 83 "send"]
+                                         :command "request"
+                                         :request-command "send"
+                                         :bot "transactor"
+                                         :command-scope-bitmask 83
+                                         :prefill [(get params :asset)
+                                                   (get params :amount)]}
+                                        constants/content-type-command-request]}
+       command-path))
+
 (deftype MessageHandler []
   Object
   (tag [this v] "c4")
   (rep [this {:keys [content content-type message-type clock-value timestamp]}]
-    (if (= content-type constants/text-content-type)
-      ;; append new content add the end, still pass content the old way at the old index
+    (condp = content-type
+      constants/text-content-type ;; append new content add the end, still pass content the old way at the old index
       #js [(:text content) content-type message-type clock-value timestamp content]
-      #js [content content-type message-type clock-value timestamp content])))
```

Comment:
@cammellos I know we had a discussion about this last time (always appending content at the end, even when not strictly necessary in command type messages), but I decided to remove it for the following reason, after discussion with @yenda :
1. As I said, most if not all the modification will be done in the content map itself, so no positional shuffling anymore (or very unlikely)
2. We want to avoid repeated content in message record whenever possible, because we fought hard to limit our bandwidth consumption and we don't want to reverse it for minor reasons like (very minor) developer convenience.
We already have no option but to do it for text messages, we don't want to do it for command messages when it's not absolutely necessary.

Reply:
@janherich As mentioned in the dm, I don't mind, I don't fully agree, but it's the team decision and I respect that

Reply:
@janherich @cammellos I was in favor of no duplication for the fix but now there is already a few versions that expect the text to be in content. If they are not able to properly read the messages received that would violate the forward compatibility policy

Reply:
@yenda The question is not about the messages of content type `text` (duplication necessary there at this time-point) but about `command` content-type, where we don't have to append content as a last item of array (as the "old" content at zero index was always a map a not interpretation change is needed there), but making it so makes the messages more regular because we can be sure that from now on, valid message content is always at the last index, opposed to:
* old style (string) content as a first and new style (map) content as a last items for the text messages
* map style content as a first item for the command messages